rocketmq-console/src/main/java/org/apache/rocketmq/console/service/impl/MessageTraceServiceImpl.java

```diff
@@ -51,4 +83,94 @@
             throw Throwables.propagate(err);
         }
     }
+
+    @Override
+    public MessageTraceGraph queryMessageTraceGraph(String key) {
+        List<MessageTraceView> messageTraceViews = queryMessageTraceKey(key);
+        return buildMessageTraceGraph(messageTraceViews);
+    }
+
+    private MessageTraceGraph buildMessageTraceGraph(List<MessageTraceView> messageTraceViews) {
+        MessageTraceGraph messageTraceGraph = new MessageTraceGraph();
+        messageTraceGraph.setMessageTraceViews(messageTraceViews);
+        if (CollectionUtils.isEmpty(messageTraceViews)) {
+            return messageTraceGraph;
+        }
+        ProducerNode producerNode = null;
+        Map<String, Map<String, Pair<MessageTraceView, MessageTraceView>>> messageTraceViewGroupMap = Maps.newHashMap();
```

Comment:
complicated definition could be defined as class

Reply:
Yes it is too complicated, already use requestIdTracePairMap instead.